python/dlisio/plumbing/errorhandler.py

```
@@ -4,10 +4,124 @@
+
+    minor:    :py:class:`ErrorHandler.Actions`
+        Action for minor specification violation
+
+    major:    :py:class:`ErrorHandler.Actions`
+        Action for major specification violation
+
+    critical: :py:class:`ErrorHandler.Actions`
+        Action for critical specification violation
+
+
+    Warnings
+    --------
+    Escaping errors is a good solution when user needs to read as much data as
+    possible, for example, to have a general overview over the file. However
+    user must be careful when using this mode during close inspection.
+    If user decides to accept errors, they must be aware that some returned data
+    will be spoiled. Most likely it will be data which is stored in the file
+    near the failure.
+
+    Warnings
+    --------
+    If user decides to accept errors, dlisio will return all the data that it
+    was able to process. But a lot of data after the failure may be skipped as
+    usually errors are not recoverable.
+
+    Warnings
+    --------
+    Be careful not to ignore too much information when investigating files.
+    dlisio might classify some issue as a minor violation, but it might be
+    critical for understanding the true level of data brokeness.
+
+    Notes
+    -----
+    Users do not have direct access to issues found during processing. They are
+    reported on user's behalf in logger or directly thrown as an exception.
+
+    Examples
+    --------
+    Define your own rules:
+
+    >>>    errorhandler = dlisio.ErrorHandler(
+    ...        info     = dlisio.ErrorHandler.Actions.SWALLOW,
+    ...        minor    = dlisio.ErrorHandler.Actions.LOG_WARNING,
+    ...        major    = dlisio.ErrorHandler.Actions.RAISE,
+    ...        critical = dlisio.ErrorHandler.Actions.RAISE)
+
+    Process a file to have a general overview:
+
+    >>> try:
+    ...    with dlisio.load(path) as fs:
+    ...        res = process(fs)
+    ...        store_results(path, res, broken=False)
```

this fictional functions through me of, and I'm a developer of this thing. They are beside the point, so just do `pass`?  I think a better illustration of the behavior is to drop the the try and let it fail, then apply custom a handler and see how everything works:

```
>>> files = dlisio.load(path)
RuntimeError: "...."

>>> handler = ErrorHandler(critical=LOG_ERROR)
>>> files = dlisio.load(path)
[ERROR] "...."
>>> for f in files:
...  pass
```

Showing that people can change behaviour midways is to unnecessarily complicate things I think. The one expert user can figure that one out for himself. But maybe add the a small docstring on the handler attribute in logical_file, that should be a hint to anyone digging around that this can be changed at any point.